awx/main/models/unified_jobs.py

```diff
@@ -981,37 +985,33 @@ def create_config_from_prompts(self, kwargs, parent=None):
+            field = None
+            # may use extra_data as a proxy for extra_vars
+            if field_name in config.SUBCLASS_FIELDS and field_name != 'extra_vars':
+                field = config._meta.get_field(field_name)
+            if isinstance(field, models.ManyToManyField):
                 many_to_many_fields.append(field_name)
                 continue
-            if isinstance(getattr(self.__class__, field_name).field, (models.ForeignKey)):
-                if value:
-                    setattr(config, "{}_id".format(field_name), value.id)
-                continue
-            key = field_name
-            if key == 'extra_vars':
-                key = 'extra_data'
-            setattr(config, key, value)
+            if isinstance(field, (models.ForeignKey)) and (value is None):
+                continue  # the null value indicates not-provided for ForeignKey case
+            setattr(config, field_name, value)
         config.save()
 
         for field_name in many_to_many_fields:
-            if field_name == 'credentials':
-                # Credentials are a special case of many to many because of how they function
-                # (i.e. you can't have > 1 machine cred)
-                job_item = set(kwargs.get(field_name, []))
-                if field_name in [field.name for field in parent._meta.get_fields()]:
-                    job_item = job_item - set(getattr(parent, field_name).all())
-                if job_item:
-                    getattr(config, field_name).add(*job_item)
+            prompted_items = kwargs.get(field_name, [])
+            if not prompted_items:
+                continue
+            if field_name == 'instance_groups':
+                # Here we are doing a loop to make sure we preserve order for this Ordered field
+                # also do not merge IGs with parent, so this saves the literal list
+                for item in prompted_items:
+                    getattr(config, field_name).add(item)
             else:
-                # Here we are doing a loop to make sure we preserve order in case this is a Ordered field
-                job_item = kwargs.get(field_name, [])
-                if job_item:
-                    parent_items = list(getattr(parent, field_name, []).all())
-                    for item in job_item:
-                        # Do not include this item in the config if its in the parent
-                        if item not in parent_items:
-                            getattr(config, field_name).add(item)
+                # Assuming this field merges prompts with parent, save just the diff
+                if field_name in [field.name for field in parent._meta.get_fields()]:
+                    prompted_items = set(prompted_items) - set(getattr(parent, field_name).all())
+                if prompted_items:
+                    getattr(config, field_name).add(*prompted_items)
```

Comment:
@john-westcott-iv I should notify you that I got sucked into rewriting more of this method. Several of things are changes I wanted to make anyway, but to recap what was going wrong...
 - We can't do `getattr(self.__class__, field_name)` when `self` is a Job and the field is `instance_groups`, because that relation doesn't exist, but since this is about saving fields to the `config`, it seems more direct to inspect the fields of _that_ object
 - When saving `config.instance_groups` we don't want the check `if item not in parent_items:`, because we don't _want_ to diff with the parent object. This is reflected in the test change.